src/understand-tidb/the-lifecycle-of-a-statement.md

```diff
@@ -0,0 +1,64 @@
+
+## AST -> Physical execution plan
+We now have a statement in AST form to handle in [handleStmt](https://github.com/pingcap/tidb/blob/30cf15a59db11c34ffe05fc926152a43327eaa61/server/conn.go#L1814)/[ExecuteStmt](https://github.com/pingcap/tidb/blob/30cf15a59db11c34ffe05fc926152a43327eaa61/session/session.go#L1620) where the Abstract Syntax Tree is [compiled](https://github.com/pingcap/tidb/blob/30cf15a59db11c34ffe05fc926152a43327eaa61/executor/compiler.go#L50) to a physical execution plan, including [optimizing](https://github.com/pingcap/tidb/blob/30cf15a59db11c34ffe05fc926152a43327eaa61/planner/optimize.go#L100) the execution plan, through a cost based optimizer. There are several steps in this process, such as name resolution, transaction management, [privilege checks](understand-tidb/privilege-management.md), handling given hints, etc.
+
+## Executing the optimized plan
+The optimized plan is executed through [runStmt](https://github.com/pingcap/tidb/blob/30cf15a59db11c34ffe05fc926152a43327eaa61/session/session.go#L1750), which builds an [executor](https://github.com/pingcap/tidb/blob/30cf15a59db11c34ffe05fc926152a43327eaa61/executor/adapter.go#L319) from the plan which will return a record set or directly execute the statements in case no records will be returned (like `INSERT`/`UPDATE`/`DELETE`). Before returning the record set, the executor starts the execution by calling the `Open()` API and the `Next()` api to retrieve the first chunk of data or execute the statement fully if no records are to be returned.
+
+### Requests sent to TiKV/TiFlash coprocessors
+
+### Sending the result back to the client
+If the statement returns a record set, it is handled in [writeChunks](https://github.com/pingcap/tidb/blob/30cf15a59db11c34ffe05fc926152a43327eaa61/server/conn.go#L1993) which loops over the record set's `Next()` 
+
+## Recommended reading:
+* [New planner design](https://github.com/pingcap/tidb/blob/master/docs/design/2018-08-29-new-planner.md)
+
+
+Agenda:
+
+* the key components that a statement needs to pass-through
+  * where the statement is received in a session - Done
+  * where the statement is parsed into an AST - Done
+  * where the AST is converted into an execution plan - Done
+  * where the execution plan is executed - Done
+  * where the request is sent to TiKV coprocessor
+  * where the execution result is returned to the client - Done
+* an example of Data Definition Language statement/DDL
+* an example of Data Modification Language statement/DML
+* an example of Data Query Language statement/DQL
+
+```sql
+-- DDL
+create table t(a bigint, b bigint);
+
+-- DML
+insert into t values (1, 1);
+
+-- DQL
+select * from t where a = 1;
+```
```

Comment:
You can take a look at the proposal of toc https://github.com/pingcap/tidb-dev-guide/pull/76 and write more special concepts of these statement in the corresponding section.

Reply:
I agree it is hard to say what is "detail" or "more special", so duplicate information a bit is always fine :)

Reply:
I just moved those examples to the subsections. I consider those subsections to be separate tasks/PRs.

Reply:
Make sense.